extensions/2.0/Khronos/KHR_materials_clearcoat/README.md

```diff
@@ -0,0 +1,96 @@
+
+## Overview
+
+This extension defines a clear coating that can be layered on top of an existing glTF material definition.  A clear coat is a common technique used in Physically-Based Rendering to represent a protective layer applied to a base material.  See [Theory, Documentation and Implementations](#theory-documentation-and-implementations)
+
+## Extending Materials
+
+The PBR clearcoat materials are defined by adding the `KHR_materials_clearcoat` extension to any compatible glTF material (excluding those listed above).  For example, the following defines a material like varnish using clearcoat parameters.
+
+```json
+{
+    "materials": [
+        {
+            "name": "varnish",
+            "extensions": {
+                "KHR_materials_clearcoat": {
+                    "clearcoatFactor": 1.0
+                }
+            }
+        }
+    ]
+}
+```
+
+### Clearcoat
+
+All implementations should use the same calculations for the BRDF inputs. Implementations of the BRDF itself can vary based on device performance and resource constraints. See [Appendix B](/specification/2.0/README.md#appendix-b-brdf-implementation) for more details on the BRDF calculations.
+
+|                                  | Type                                                                            | Description                            | Required             |
+|----------------------------------|---------------------------------------------------------------------------------|----------------------------------------|----------------------|
+|**clearcoatFactor**               | `number`                                                                        | The clearcoat layer intensity.         | No, default: `0.0`   |
+|**clearcoatTexture**              | [`textureInfo`](/specification/2.0/README.md#reference-textureInfo)             | The clearcoat layer intensity texture. | No                   |
+|**clearcoatRoughnessFactor**      | `number`                                                                        | The clearcoat layer roughness.         | No, default: `0.0`   |
+|**clearcoatRoughnessTexture**     | [`textureInfo`](/specification/2.0/README.md#reference-textureInfo)             | The clearcoat layer roughness texture. | No                   |
+|**clearcoatNormalTexture**        | [`normalTextureInfo`](/specification/2.0/README.md#reference-normaltextureinfo) | The clearcoat normal map texture.      | No                   |
+  
+The clearcoat formula `f_clearcoat` is computed using the specular term from the glTF 2.0 Metallic-Roughness material, defined in [Appendix B](/specification/2.0/README.md#appendix-b-brdf-implementation).  Use F0 equal `0.04`, base color white `1.0, 1.0, 1.0`, metallic value `0.0`, and the clearcoat roughness value defined in this extension as follows:
+
+```
+clearcoatRoughness = clearcoatRoughnessFactor * clearcoatRoughnessTexture.g
```

Comment:
Given the [challenges](https://github.com/KhronosGroup/glTF/issues/1682) of compressed textures, do we want to continue requiring that clearcoat be packed into `.r` and clearcoat roughness into `.g`? I'm all for the memory savings, but we'll need a workaround in `KHR_texture_basisu` almost right out of the gate, and we haven't established whether that adjustment belongs in this extension or `KHR_texture_basisu`.

Requiring `.r` and `.a` would be one option. Throwing up our hands and dealing with this packing stuff once swizzle is generally available is another, and increasingly tempting. 😞

Finally – I'd missed this line on the initial readthroughs. I would consider mentioning the packing in the table above, in addition to the sample code.

Reply:
What's wrong with using RG? As of today, BC5 is available in Chrome Canary on virtually all desktops.

Reply:
Great news, but that's the whole conversation around https://github.com/KhronosGroup/glTF/issues/1682#issuecomment-581608529, isn't it? We can transcode two independent ETC1S slices to `xy01` on desktop, but not on mobile, which would still require `xxxy`?

Reply:
On mobile, two ETC1S slices are used as two ETC1 (two PVRTC on iOS for now) textures or one EAC RG11. There's no need to do any swizzling there.

Reply:
Ah I see. We can use two PVRTC textures here because implementations already have to support separated clearcoatTexture/clearcoatRoughnessTexture, whereas loading a normal map's channels from two PVRTC textures would be a new requirement on implementations.

That's fine with me, then.